axolotls/string_column.py

```diff
@@ -0,0 +1,102 @@
+import string
+from .column_base import ColumnBase
+from . import dtypes as dt
+from typing import Optional
+import torch
+
+class StringColumn(ColumnBase):
+    def __init__(
+        self, 
+        values: torch.ByteTensor, 
+        offsets: torch.Tensor,
+    ) -> None:
+        # String Column doesn't support null value for now
+        # It is straightforward to support it (with extra optional), but hold on for simplicity, 
+        #   and only add it once there is clear use cases.
+        super().__init__(dtype=dt.string)
+
+        if not isinstance(values, torch.ByteTensor) or values.dim() != 1:
+            raise ValueError("StringColumn expects values to be 1D ByteTensor")
+        if not isinstance(offsets, torch.Tensor) or offsets.dim() != 1 or offsets.dtype != torch.int32:
+            raise ValueError("StringColumn expects 1D offsets int32 Tensor")
+
+        self._values = values
+        self._offsets = offsets
+
+    def clone(self) -> "StringColumn":
+        return StringColumn(
+            values=self.values.detach().clone(),
+            offsets=self.offsets.detach().clone(),
+        )
+
+    def __getitem__(self, key) -> str:
+        if isinstance(key, int):
+            bytes: torch.ByteTensor = self.values[self.offsets[key] : self.offsets[key + 1]]
+            return bytearray(bytes).decode('utf-8')
+
+        if isinstance(key, slice):
+            # non-continugous slice is not supported yet
+            assert key.step is None
+            start = key.start or 0
+            stop = key.stop or len(self)
+
+            values = self.values[self.offsets[start] : self.offsets[stop]]
+
+            # convert offsets to length for selection
```

Comment:
this can be simplified , e.g. something like 
```
offsets = self.offsets[start : stop]
offsets -= offsets[0]
```

Will leave the simplification not in this PR